examples/ImageSearchLocalization/app.py

```diff
@@ -0,0 +1,211 @@
+from PIL import Image, ImageOps
+import streamlit as st
+from tempfile import NamedTemporaryFile
+import validators
+import requests
+import os
+import urllib3
+urllib3.disable_warnings()
+
+from marqo import Client
+
+client = Client()
+
+def load_image_from_path(image):
+    """loads an image into PIL from a string path that is
+    either local or a url
+    Args:
+        image (str): _description_
+    Raises:
+        ValueError: _description_
+    Returns:
+        ImageType: _description_
+    """
+    
+    if os.path.isfile(image):
+        img = Image.open(image)
+    elif validators.url(image):
+        img = Image.open(requests.get(image, stream=True).raw)
+    else:
+        raise ValueError(f"input str of {image} is not a local file or a valid url")
+
+    return img    
+
+def render_images(images, captions=None, boxes=None):
+    """renders a list of image pointers
+
+    Args:
+        images (_type_): _description_
+        captions (_type_, optional): _description_. Defaults to None.
+    """
+    
+    if boxes is None:
+        images = [ImageOps.expand(load_image_from_path(image), border=100, fill=(255,255,255) ) for image in images]
+    else:
+        images = [ImageOps.expand(draw_box(load_image_from_path(image), box), border=100, fill=(255,255,255) ) for box,image in zip(boxes,images)]
+        
+
+    if captions is None:
+        st.image(images, use_column_width=False, width=230)
+    else:
+        st.image(images, caption=captions, use_column_width=False)#, width=230)
+
+
+def load_image(image_file):
+    """loads an image using PIL
+
+    Args:
+        image_file (_type_): _description_
+
+    Returns:
+        _type_: _description_
+    """
+    img = Image.open(image_file)
+    return img
+
+def draw_box(img, box):
+    import math
+    from PIL import Image, ImageDraw
+        
+    # create rectangle image
+    img1 = ImageDraw.Draw(img)  
+    img1.rectangle(box, outline  = 'orange', width = 8)
+    return img
+
+def main():
+
+    data_dir = "/home/jesse/code/stable-diffusion/scripts/coco/"
```

Comment:
Probably change this to a placeholder, and add instructions (or a link to instructions) on how to set up the data dir

Reply:
Thanks. Yeah, the demo is still a WIP